pkg/cli/cli.go

```diff
@@ -202,6 +213,27 @@ func runListHostsCommand(c *cobra.Command, _ []string, opts *subCommandOptions)
 	return nil
 }
 
+func runDeleteHostsCommand(c *cobra.Command, args []string, opts *subCommandOptions) error {
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var errs error
+	for _, arg := range args {
+		wg.Add(1)
+		go func(name string) {
+			defer wg.Done()
+			url := opts.BaseURL + "/hosts/" + name
+			if err := doRequest(opts.HTTPClient, "DELETE", url, nil, nil); err != nil {
+				mu.Lock()
+				errs = multierror.Append(errs, fmt.Errorf("delete host %q failed: %w", name, err))
+				defer mu.Unlock()
```

Comment:
Either move this line to right after the call to `Lock()` or don't use defer (if you know nothing will return in between lock and unlock). By the time control reaches this line there is no need to defer anymore

Reply:
good catch! done